pkg/azure/ipam/node.go

```diff
@@ -175,3 +175,11 @@ func (n *Node) ResyncInterfacesAndIPs(ctx context.Context, scopedLog *logrus.Ent
 
 	return available, nil
 }
+
+// GetMaximumAllocatableIPv4 returns the maximum amount of IPv4 addresses
+// that can be allocated to the instance
+// TODO: If this is necessary for Azure, need to figure out how to get or
```

Comment:
The limits for Azure are 256 across a node (also 256 on a NIC). So in total you can every combination but in total across all NICs on a node needs to be max 256.

Reply:
@ungureanuvladvictor @mvisonneau do you want to address this in this PR or do you think it's more appropriate to split out into a separate effort to enable this functionality on the Azure side?

Reply:
I know this went through many iterations so I approved it so it does not stay in the queue much time.

If @mvisonneau wants to fix this I'm happy to take another look.

Reply:
oh thanks for the info, I will update it 👍

Reply:
done!